### DISCUSSION TITLE
Flows: How to create a for-loop?

### DISCUSSION THREAD

**Discussion by erondpowell** [directus/directus#16011] [Feature Requests]
**Ref:** https://discord.com/channels/725371605378924594/1030485118722261053

**^^TLDR:** 
If you want to trigger a flow once for each item in a multi-item payload, you have to pass the payload to another flow... Except in item.create event hooks (any maybe others) which trigger the flow once for each item created.

I realize for create item event hooks, it might be hard/impossible to have the flow trigger once for all items created.
But for event hooks triggers, manual triggers and webhook triggers (as well as `Another Flow` triggers) with an array-like, multi-item payload. It would be *super nice* if we had a toggle option to either run the flow once for the entire payload, or once for each item in the payload.

**Comment by nickrum** [comment]
> But for event hooks triggers, manual triggers and webhook triggers (as well as `Another Flow` triggers) with an array-like, multi-item payload. It would be _super nice_ if we had a toggle option to either run the flow once for the entire payload, or once for each item in the payload.

That might also be a little tricky to do because similar to how the event system is hard-coded to trigger once for each item created, it is also hard coded to trigger for all updated items at once. We would either have to hack around that or change the behavior for both flows and hook extensions as a breaking change in v10.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
